src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLSPTextViewConnectionListener.cs

```diff
@@ -111,31 +113,36 @@ public void SubjectBuffersConnected(ITextView textView, ConnectionReason reason,
             // Initialize the user's options and start listening for changes.
             // We only want to attach the option changed event once so we don't receive multiple
             // notifications if there is more than one TextView active.
-            if (_razorEditorOptions == null && textView.TextBuffer.IsRazorLSPBuffer())
+            if (!textView.TextBuffer.Properties.ContainsProperty(RazorEditorOptions) &&
+                textView.TextBuffer.IsRazorLSPBuffer())
             {
-                _razorEditorOptions = _editorOptionsFactory.GetOptions(textView);
-                Assumes.Present(_razorEditorOptions);
+                var razorEditorOptions = _editorOptionsFactory.GetOptions(textView);
+                Assumes.Present(razorEditorOptions);
+                textView.TextBuffer.Properties[RazorEditorOptions] = razorEditorOptions;
+                _textBuffer = textView.TextBuffer;
+
                 RazorOptions_OptionChanged(null, null);
-                _razorEditorOptions.OptionChanged += RazorOptions_OptionChanged;
+                razorEditorOptions.OptionChanged += RazorOptions_OptionChanged;
             }
         }
 
         public void SubjectBuffersDisconnected(ITextView textView, ConnectionReason reason, IReadOnlyCollection<ITextBuffer> subjectBuffers)
         {
             // When the TextView goes away so does the filter.  No need to do anything more.
             // However, we do need to detach from listening for option changes to avoid leaking.
-            if (_razorEditorOptions != null)
+            if (_textBuffer != null && _textBuffer.Properties.TryGetProperty(RazorEditorOptions, out IEditorOptions editorOptions))
             {
-                _razorEditorOptions.OptionChanged -= RazorOptions_OptionChanged;
-                _razorEditorOptions = null;
+                editorOptions.OptionChanged -= RazorOptions_OptionChanged;
+                _textBuffer.Properties.RemoveProperty(RazorEditorOptions);
```

Comment:
This will break if there's multiple ITextViews open I believe. Aka, if you have 2 and close 1 this will stop tracking the text buffer before the remaining 1 is closed.

You'll probably need to reference count the ITextViews and attach/detatch from settings accordingly. Or if there's a way to track settings at the ITextBuffer layer then maybe we could get out of the business of ITextViews entirely? Lets chat more tomorrow, this brings up some really interesting interactions with LSP + VS that I don't know if the platform team has even considered yet. Paving the way!